### PR TITLE
Set up UserTag -> Profile nav

### PR DESCRIPTION
**Pull request by tylerhm**
# Changes
Pressing the `UserTag` now navigates to the respective user's profile.

Edit: After applying the (fantastic) changes recommended by @abrugal, this has basically become an entirely different PR. This is a complete re-work of how navigation is handled in the application.

1. Any screen that can be displayed from any tab has now been moved to the `tabGlobal` routes folder. Every tab's routes extend the routes here.
2. A new set of `Composite<Screen/Navigation>Prop(s)` has been created, and must be used when navigating to screens that are global.
3. Several locations that passed a `navigation` prop have been swapped out for the `useNavigate` hook with the global navigation props typing.

_Note: The Leaderboard screens have not been moved to the global route collection because the need has not yet arisen, and this PR is already big enough_

## Disclaimer
It is very possible that I missed some files when swapping over, so I'd request a pretty thorough review + local testing on your end to make sure things aren't broken.

# Issue ticket number and link
#115

### CODE REVIEW

**Comment by abrugal** [comment]
Do we want it to travel to the Profile tab though? Like wouldn't it be jarring to be on the home page, click a user tag and then be taken to the profile tab? 

To fix this I was thinking that we add a Profile screen for each tab that could take you to a user's profile. You would basically just have to make a wrapper for the profile component, and add the screen name to the paramList and all that. Also would have to add screen name for Followers in the tab too. The actual components wouldn't need to be changed, since it's already set up so that we pass in the navigation function as a prop.

**Comment by tylerhm** [comment]
Hmmmm I did think about this, and landed on navigating to the `Profile` tab but I think you are right. I'll change this before merging for sure.

**Comment by abrugal** [comment]
Sounds good. If you need help with this lmk

**Comment by tylerhm** [comment]
Thanks! I actually had it implemented per tab and then swapped to profile because I thought I liked it better lol. You're definitely right though.

File: components/profile/Profile.tsx [hunk 1]
```diff
@@ -96,14 +99,16 @@ const Profile = ({ profileIsMine, userOfProfile, navigate }: Props) => {
...
             </Button>
             <Center>
               <Pressable
-                onPress={() =>
-                  navigate('Follows', { username: userOfProfile.username })
-                }
+                onPress={async () => {
+                  navigation.navigate('Follows', {
+                    username: await userOfProfile.getUsername(),
+                  });
```
Comment (abrugal) [inline components/profile/Profile.tsx:110]: This should be using navigation.push

**Review by abrugal — COMMENTED** [review]
Left a comment in Profile.tsx. Also I tested this branch and when going to the active routes tab and pressing the route the app crashes.  

It says:
Error while updating property 'flexGrow'
String cannot be cast to Double

**Comment by tylerhm** [comment]
Hmmm that `flexGrow` error sounds like its from a different commit. I didn't touch any styling in this one. Addressing the accidental `.navigate` now.

**Comment by tylerhm** [comment]
> Left a comment in Profile.tsx. Also I tested this branch and when going to the active routes tab and pressing the route the app crashes.

Just confirmed that this bug is in `main`. For some reason it doesn't throw the error on my machine? Putting a new PR up to address

File: ' [hunk 2]
```diff
@@ -0,0 +1,166 @@
+import ProfileBanner from '../../components/profile/ProfileBanner';
```
[the remark this answers is not in the document]
Comment (tylerhm) [reply ':1]: bruh wtf lmao, I'll take a look. Thanks for checking so fine grained here, stuff like this slips through with massive commits like this.

**Review by abrugal — APPROVED** [review]
Great work!

**Pull request merged**